src/mmfatl.c

```diff
@@ -0,0 +1,1591 @@
+/*****************************************************************************/
+/*        Copyright (C) 2022  Wolf Lammen                                    */
+/*            License terms:  GNU General Public License                     */
+/*****************************************************************************/
+/*34567890123456 (79-character line to adjust editor window) 2345678901234567*/
+
+/*!
+ * \file mmfatl.c a self-contained set of text printing routines using
+ * dedicated pre-allocated memory, designed to likely run even under difficult
+ * conditions (corrupt state, out of memory).
+ */
+
+#include "mmfatl.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+
+#ifdef TEST_MMFATL
+    // We intercept malloc/free to test proper
+    // freeing after allocation of memory
+    static void* MALLOC(size_t size);
+    static void FREE(void* ptr);
+#else
+#   define FREE free
+#   define MALLOC malloc
+#endif
+
+/*!
+ * C has no boolean type, instead an int is used.  To ease code reading we
+ * encode one boolean value here.
+ * int cond = FALSE; if (cond) { } skips the code block.
+ */
+#define FALSE 0
```

Comment:
What the heck is this? Don't make macros for true and false, use the appropriate imports and use `true` and `false`.

Reply:
I don't know how to interpret https://en.cppreference.com/w/c/types wrt to bool/true/false.  For one they are C99, and then they look deprecated.  But I can certainly remove the macros here, if considered necessary.

Reply:
They are hardly deprecated; to the contrary the reason they are going away in C23 is because the *macro* is going away to make way for [the keyword](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2655.pdf) with the same semantics. C has clearly taken steps toward making `bool` / `true` / `false` the preferred spelling and `stdint.h` is how you should use it in C11 (which we are using).

Reply:
I did a quick lookup at the time of writing, and decided to use my macros as a replacement.  Looking closer now I too think we should use the standard keywords.